### PR TITLE
 Update m3cluster and m3metrics

### PR DESCRIPTION
**Pull request by cw9**

### CODE REVIEW

File: aggregator/aggregator_mock.go [hunk 1]
```diff
@@ -18,340 +18,396 @@
...
 	"context"
+	"reflect"
 	"time"
 
 	"github.com/m3db/m3aggregator/generated/proto/flush"
 	"github.com/m3db/m3cluster/placement"
 	"github.com/m3db/m3cluster/shard"
 	"github.com/m3db/m3metrics/metric/aggregated"
 	"github.com/m3db/m3x/watch"
+	"github.com/uber-go/tally"
```
Comment (xichen2020) [inline aggregator/aggregator_mock.go]: Hey so this is not ordered properly due to an issue I fixed in the latest rev of ci-scripts, can you pull that in and re-run mockgen? Thanks.

File: integration/custom_aggregation_test.go [hunk 2]
```diff
@@ -26,16 +26,17 @@ import (
...
 
 	"github.com/stretchr/testify/require"
 )
 
 var (
-	compressor                             = policy.NewAggregationIDCompressor()
-	compressedMin, _                       = compressor.Compress(policy.AggregationTypes{policy.Min})
-	compressedMinAndMax, _                 = compressor.Compress(policy.AggregationTypes{policy.Min, policy.Max})
+	compressor                             = aggregation.NewIDCompressor()
+	compressedMin, _                       = compressor.Compress(aggregation.Types{aggregation.Min})
```
Comment (xichen2020) [inline integration/custom_aggregation_test.go]: Can you change this and the following one to use `MustCompress`? Thanks

File: services/m3aggregator/config/aggregator.go [hunk 3]
```diff
@@ -369,7 +371,7 @@ func (c *streamConfiguration) NewStreamOptions(instrumentOpts instrument.Options
 }
 
 type placementManagerConfiguration struct {
-	KVConfig         kv.Configuration               `yaml:"kvConfig"`
+	KVConfig         kv.OverrideConfiguration       `yaml:"kvConfig"`
```
Comment (xichen2020) [inline services/m3aggregator/config/aggregator.go]: Do we need to set loggers here as well?

File: aggregator/aggregator_mock.go (hunk 1 above)
Comment (cw9) [reply aggregator/aggregator_mock.go]: sounds good

File: integration/custom_aggregation_test.go (hunk 2 above)
Comment (cw9) [reply integration/custom_aggregation_test.go]: sounds good

File: services/m3aggregator/config/aggregator.go (hunk 3 above)
Comment (cw9) [reply services/m3aggregator/config/aggregator.go]: Nope, I think normal logger here is good enough, for flush times mgr we need to config to warn level logger

**Pull request merged**